packages/stellar-axelar-std-derive/src/contractimpl.rs

```diff
@@ -9,49 +8,45 @@ use syn::{
 use crate::utils::{parse_env_identifier, PrependStatement};
 
 pub fn contractimpl(impl_block: &mut ItemImpl) -> Result<proc_macro2::TokenStream, syn::Error> {
-    // this needs to be defined before the iteration, because during it, we can't get a reference to the impl block
-    let any_stateful_endpoints = any_stateful_endpoints(impl_block);
-
-    impl_block
-        .items
-        .iter_mut()
-        .filter_map(any_stateful_endpoints)
-        .chunk_by(is_allowed_during_migration)
-        .into_iter()
-        .try_for_each(|(is_allowed, mut method)| {
-            if is_allowed {
-                // if this the attribute is not removed, the compiler will try to resolve it,
-                // and it will need to be defined as a standalone attribute macro
-                method.for_each(remove_allow_during_migration_attribute);
-            } else {
-                method.try_for_each(block_during_migration)?
+    for method in impl_block.items.iter_mut() {
+        if let ImplItem::Fn(method_fn) = method {
+            let is_contract_endpoint =
+                impl_block.trait_.is_some() || matches!(method_fn.vis, Visibility::Public(_));
+            let is_stateful = has_args(&method_fn);
```

Comment:
the issue isn't really that the fn has arguments, but that we can restrict the code gen to fns that have env as a parameter. I split that previously as an optimization to filter more before doing the more expensive parsing, but now, the env parsing is actually done multiple times instead, so there is no benefit to that check.
I suggest that you try to parse the env_param here (either with `if let Ok(env) = parse_env_identifier(...)` or with `parse_env_identifier(...).and_then(...)`) and then pass it into `block_during_migration` and `instance_ttl_extension` instead of parsing it three times